### PR TITLE
Animation options parser

### PR DESCRIPTION
**Pull request by frankdiox**
@argelius @anatoo This PR is finished, please review it when you have some time.

`animation-options="{delay: 1, duration: 1, timing: 'ease-in'}"` is currently broken because it's not a valid json for `JSON.parse( ... )`. 

This supports things we don't currently use in `animation-options`, such as arrays and nested objects.

### CODE REVIEW

File: core/lib/animationOptionsParser.spec.es6 [hunk 1]
```diff
@@ -0,0 +1,23 @@
+describe('ons._animationOptionsParser', () => {
+  it('exists', () => {
+    expect(ons._animationOptionsParser).to.be.ok;
+  });
+
+  it('accepts valid strings', () => {
+    expect(ons._animationOptionsParser.parse('{}')).to.be.ok;
+    expect(ons._animationOptionsParser.parse('{ _v: false,  $qw : .1, w$_ : {x : 1e3}, as :\' a\', _2  : [6, {x : \'%^&\'}, 8] ,  q:{x:{p:[1,2]}} , xc:"bnm"   }')).to.be.ok;
+    expect(ons._animationOptionsParser.parse('[]')).to.be.ok;
+    expect(ons._animationOptionsParser.parse('[{ i: 1},3,  "qw",\'zx\', 1e2]')).to.be.ok;
```
Comment (argelius) [inline core/lib/animationOptionsParser.spec.es6]: Should this be valid? I think it should only accept objects.

File: core/lib/animationOptionsParser.spec.es6 [hunk 2]
```diff
@@ -0,0 +1,23 @@
+describe('ons._animationOptionsParser', () => {
+  it('exists', () => {
+    expect(ons._animationOptionsParser).to.be.ok;
+  });
+
+  it('accepts valid strings', () => {
```
Comment (argelius) [inline core/lib/animationOptionsParser.spec.es6]: Can you also write tests that confirm the integrity of the objects?

**Comment by argelius** [comment]
@frankdiox Can you also write tests that confirm that the regression is fixed? I mean, test that the attribute actually works for some element.

**Comment by frankdiox** [comment]
@argelius Done!

**Comment by argelius** [comment]
Thanks @frankdiox !

**Comment by frankdiox** [comment]
@argelius Conflict solved.

**Pull request merged**